src/System.Runtime.Analyzers/Core/UseOrdinalStringComparison.cs

```diff
@@ -42,99 +45,147 @@ public override void Initialize(AnalysisContext analysisContext)
-                            return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
-                                methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Int32 &&
-                                methodSymbol.Parameters[2].Type.SpecialType == SpecialType.System_String &&
-                                methodSymbol.Parameters[3].Type.SpecialType == SpecialType.System_Int32 &&
-                                methodSymbol.Parameters[4].Type.SpecialType == SpecialType.System_Int32 &&
-                                methodSymbol.Parameters[5].Type.Equals(stringComparisonType);
-                    }
+                    return; 
                 }
-                else if (string.Equals(methodSymbol.Name, EqualsMethodName, StringComparison.Ordinal))
+                reportDiagnostic(Diagnostic.Create(Rule, GetOperatorTokenLocation(operation.Syntax)));
+            }
+        }
+
+        private static bool IsEqualsOrCompare(string methodName)
+        {
+            return string.Equals(methodName, EqualsMethodName, StringComparison.Ordinal) ||
+                string.Equals(methodName, CompareMethodName, StringComparison.Ordinal);
+        }
+
+        private static bool IsAcceptableOverload(IMethodSymbol methodSymbol, INamedTypeSymbol stringComparisonType)
+        {
+            return methodSymbol.IsStatic
+                ? IsAcceptableStaticOverload(methodSymbol, stringComparisonType)
+                : IsAcceptableInstanceOverload(methodSymbol, stringComparisonType);
+        }
+
+        private static bool IsAcceptableInstanceOverload(IMethodSymbol methodSymbol, INamedTypeSymbol stringComparisonType)
+        {
+            if (string.Equals(methodSymbol.Name, EqualsMethodName, StringComparison.Ordinal))
+            {
+                switch (methodSymbol.Parameters.Length)
                 {
-                    switch (methodSymbol.Parameters.Length)
-                    {
-                        case 2:
-                            // (object, object) is acceptable
-                            return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_Object &&
-                                methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_Object;
-                        case 3:
-                            // (string, string, StringComparison) is acceptable
-                            return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
-                                methodSymbol.Parameters[1].Type.SpecialType == SpecialType.System_String &&
-                                methodSymbol.Parameters[2].Type.Equals(stringComparisonType);
-                    }
+                    case 1:
+                        // the instance method .Equals(object) is OK
+                        return methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_Object;
+                    case 2:
```

Comment:
For consistency with what you did in `IsAcceptableStaticOverload` (where all of the cases had a comment), consider adding comment: `// .Equals(string, System.StringComparison) is acceptable.`